Java/core/src/main/java/com/amazon/randomcutforest/AbstractForestUpdateExecutor.java

```diff
@@ -59,9 +70,9 @@ public long getTotalUpdates() {
      */
     public void update(double[] point) {
         double[] pointCopy = cleanCopy(point);
-        P updateInput = updateCoordinator.initUpdate(pointCopy);
-        List<P> results = update(updateInput);
-        updateCoordinator.completeUpdate(updateInput, results);
+        P updateInput = updateCoordinator.initUpdate(pointCopy, currentIndex);
```

Comment:
The pointCoordinator does not need to know the sequence information, for example it can be a simple passthrough class (say for double[]). The pointCoordinator is basically a repository that does not know time/sequence information.